Написание рекомендации

s3nkyh opened the pull request.

Commit:
added: RecomContrl, RecomDto, RecomService, SkillOfferDto

Commit:
бааааааааааааааааааааааааги

Commit:
реализовал механику добавления рекомендации

Commit:
удалил recommendationId из SkillOfferDto

Commit:
--

Commit:
поправил RecommendationController

Commit:
-

Commit:
Subscribe to a user

Commit:
добавил: ValidationRecommendationUtils, SkillDto
исправил: RecommendationMapper, RecommendationController, RecommendationService,
 RecommendationDto,
 SkillOfferDto

Commit:
исправил импорты

Commit:
исправил SkillOfferRepository и RecommendationService

Commit:
Unsubscribe from the user

Commit:
--

Commit:
исправил RecommendationMapper
удалил SkillDto.java

Commit:
Get all subscribers

Commit:
Get the number of subscribers

Commit:
Get the count of followee

Commit:
исправил RecommendationMapper

Commit:
убрал лишние аннотации

Commit:
убрал лишние аннотации

Commit:
-

Commit:
исправил RecommendationMapper
добавил метод deleteAllAndCreate в RecommendationService

Commit:
сохранил список гарантов

Commit:
реализовал механику удаления рекомендации

Commit:
реализовал механику получение всех данных рекомендации

Commit:
переименовал: mapper -> recommendationMapper,
toEntity -> toRecommendation и др.
вынес все сообщения в константы и сделал их русскими

Commit:
исправил маппер и валидацию

Commit:
исправил маппер и валидацию

Commit:
добавил логи и проверку на null

Commit:
реализовал механику получения всех данных рекомендации

Commit:
добавил проверку на null

Commit:
исправил RecommendationService(не до конца), SkillOfferRepository, ValidationRecommendationUtils

Commit:
исправил RecommendationService,
RecommendationValidation
добавил метод в RecommendationRepository

Commit:
исправил RecommendationRepository

Commit:
исправил RecommendationService,
RecommendationValidation
добавил метод в RecommendationRepository и SkillOfferRepository

Commit:
-

Commit:
Испаврил RecommendationMapper
удалил UserDto

Commit:
Испаврил RecommendationMapper

Commit:
исправил RecommendationService,
RecommendationValidation,
RecommendationMapper
добавил метод в RecommendationRepository и SkillOfferRepository

Commit:
исправил RecommendationService,
RecommendationValidation,
RecommendationMapper
добавил метод в RecommendationRepository и SkillOfferRepository

Commit:
Refactored

jibulani committed:
Merge pull request #2640 from CorporationX/feature-BJS2-65051

Subscribe to other users

Commit:
Написание рекомендации

Реализовал основные операции: дать рекомендацию, обновить рекомендацию, удалить рекомендацию, получить все рекомендации пользователя и получить все данные рекомендации.

Commit:
добавил ветку с реализацией удаления

Commit:
добавил ветку с реализацией получения всех данных рекомендации

Commit:
добавил ветку с реализацией получения всех рекомендаций пользователя

Commit:
добавил SkillOfferMapper, билдер в createSkillOffer в RecommendationService

Commit:
добавил билдер в deleteAllAndCreate в RecommendationService

SergeyErmoshenko commented:
Ветку на мастер команды смени и конфликты решить надо

SergeyErmoshenko commented on target/slf4j/file/app.log:
```diff
@@ -0,0 +1,97 @@
+
+Action:
+
+Consider defining a bean of type 'java.lang.String' in your configuration.
```
Логи из индекса удали, и обнови мастер. Там запушены логи в gitignore

SergeyErmoshenko commented on src/main/java/school/faang/user_service/controller/RecommendationController.java:
```diff
@@ -0,0 +1,35 @@
+    }
+
+    public RecommendationDto updateRecommendation(RecommendationDto recommendationDto) {
+        return recommendationService.update(recommendationDto);
```
Можно логов о начале и завершении добавить

SergeyErmoshenko commented on src/main/java/school/faang/user_service/service/RecommendationService.java:
```diff
@@ -0,0 +1,158 @@
+    public void delete(Long id) {
+        if (id == null) {
+            throw new DataValidationException("Id не должен быть null");
+        }
```
Вроде бы договорились логи на английском писать

Commit:
добавил логи на обновление рекомендации,
поменял логи на английский

Commit:
-

PavelEzhkov commented on src/main/java/school/faang/user_service/controller/RecommendationController.java:
```diff
@@ -0,0 +1,40 @@
+    }
+
+    public RecommendationDto updateRecommendation(RecommendationDto recommendationDto) {
+        log.info("Starting update");
```
если уж добавлять логи то надо их было добавить во все методы

PavelEzhkov commented on src/main/java/school/faang/user_service/service/RecommendationService.java:
```diff
@@ -0,0 +1,158 @@
+        if (authorId == null) {
+            throw new DataValidationException("The author's ID cannot be null");
+        }
+        Pageable pageable = PageRequest.of(0, 10);
```
Так ты получишь только 10 записей

PavelEzhkov commented on src/main/java/school/faang/user_service/service/RecommendationService.java:
```diff
@@ -0,0 +1,158 @@
+            throw new DataValidationException("The author's ID cannot be null");
+        }
+        Pageable pageable = PageRequest.of(0, 10);
+        Page<Recommendation> recommendationPage = recommendationRepository.findAllByAuthorId(authorId, pageable);
+        List<Recommendation> recommendationList = recommendationPage.getContent();
```
Можно сделать одной строкой

PavelEzhkov commented on src/main/java/school/faang/user_service/service/RecommendationService.java:
```diff
@@ -0,0 +1,158 @@
+    public RecommendationDto create(RecommendationDto recommendationDto) {
+        RecommendationValidation.validateRecommendationContent(recommendationDto.getContent());
+        RecommendationValidation.validateSkills(recommendationDto, skillOfferRepository.findAllSkillOffers());
+        LocalDateTime lastRecommendation = recommendationRepository.
```
Я бы вынес в приватный метод получение последней рекомендации

PavelEzhkov commented on src/main/java/school/faang/user_service/service/RecommendationService.java:
```diff
@@ -0,0 +1,158 @@
+
+        Recommendation recommendation = recommendationMapper.toRecommendation(recommendationDto);
+
+        Long recommendationId = createRecommendation(recommendation);
```
Давай сделаем так что createRecommendation(recommendation); будет возвращать Recommendation уже с id

Commit:
Добавил логи во все методы контроллера.
Исправил Pageable.
Вынес повторяющиеся строки кода в приватные методы.
Исправил метод createRecommendation

Commit:
сделал все методы RecommendationValidation не статическими

PavelEzhkov committed:
Merge branch 'dev' into feature/BJS2-65857-writting-recommendation

s3nkyh committed:
Покрыть код задачи "Написание рекомендации" unit-тестами (#2819)

* Покрыл код задачи "Написание рекомендации" unit-тестами,
также немного поменял логику в RecommendationService

* Исправил аннотации в RecommendationController,
Добавил аннотации в дто,
Вынес повторяющийся код в @BeforeEach

* добавил @RequestMapping("/recommendation")

---------

Co-authored-by: Yaroslav Olkin <shrelok.pupkin@yandex.ru>

Commit:
Merge remote-tracking branch 'origin/medusa-master-stream9' into feature/BJS2-65857-writting-recommendation

# Conflicts:
#	build.gradle.kts
#	src/main/java/school/faang/user_service/controller/SubscriptionController.java
#	src/main/java/school/faang/user_service/dto/UserDto.java
#	src/main/java/school/faang/user_service/dto/UserFilterDto.java
#	src/main/java/school/faang/user_service/service/SubscriptionService.java
